gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/oauth/OAuthViewModel.kt

```diff
@@ -28,14 +30,21 @@ internal class OAuthViewModel(
     val actions = _actions.receiveAsFlow()
 
     init {
+        startOAuth()
+    }
+
+    fun startOAuth() {
         viewModelScope.launch {
             _actions.send(OAuthAction.StartOAuth)
+            _uiState.update { currentState ->
+                currentState.copy(status = OAuthStatus.LoginRequired)
```

Comment:
Ohh it's already a default state so no need for that. I think you're setting it after the user taps the login if an email was wrong, right? I think I would keep the `WrongEmailAuthorized` status in case the user decides to close the browser to preserve that error message.

Reply:
> I think you're setting it after the user taps the login if an email was wrong, right?

Exactly.

> I think I would keep the WrongEmailAuthorized status in case the user decides to close the browser to preserve that error message.

I implemented that way because I thought about it like:

1.- The email was wrong
2.- The user sees the error and clicks the login button again
3.- The error is no longer valid; the user has seen it and opened the flow again. 

I'll ask iOS folk what they are doing.

Reply:
sure! Either way is okay I guess. I will approve then!

Reply:
👍 I'll merge it, and if we decide on a different thing, I'll open another PR.